### DISCUSSION TITLE
Javascript in Column Formulas

### DISCUSSION THREAD

**Discussion by jorge-gbs** [Budibase/budibase#14502] [Help] [answered]
Hello, I am trying to add some javascript to make a formula based on if certain fields are completed. this always errors out. What am I doing wrong?

```javascript
function checkTitle () {
  if (typeof $("Title") === "string" && $("Title").length === 0) {
    return $("Format")+" - "+$("Platform")+" - "
  } else if (str === null) {
    return $("Format")+" - "+$("Platform")+" 2 "
  } else {
    return $("Format")+" - "+$("Platform")+" 3 "
  }
}
//return $("Format")+" - "+$("Platform")+" 3 "
return checkTitle();```

**Comment by ConorWebb96** [comment, marked as answer]
`else if (str === null) {`

You haven't defined a variable called `str` anywhere.

**Reply by jorge-gbs** [reply]
That was it, anyway to get line numbers in the error output? That would be great.

**Comment by ashishkumarpalai** [comment]
Here's a step-by-step breakdown of what might be going wrong in your code and how you can fix it:

### 1. **Problem with `$("Title")` Selector:**
   - **Issue:** The code uses `$("Title")`, which suggests you're trying to select an element with an ID of `Title`. However, jQuery's `$` function typically selects elements with `$("#Title")` (for IDs) or `$(".Title")` (for classes). Using just `"Title"` might not be the correct way to select an element.
   - **Solution:** If you're trying to select an element with the ID `Title`, you should change `$("Title")` to `$("#Title")`.

### 2. **Checking the Length of the Element:**
   - **Issue:** The code checks `$("Title").length === 0` to determine if the string is empty. If `$("Title")` is a jQuery object (i.e., it’s selecting an element), you should be checking the `val()` of the element to see if it is empty.
   - **Solution:** Modify the condition to `$("#Title").val().length === 0` to check if the input field is empty.

### 3. **The `str` Variable is Not Defined:**
   - **Issue:** In the second `else if`, the code checks `str === null`, but `str` is not defined anywhere in your code. This will cause an error.
   - **Solution:** If you intended to use a variable called `str`, you need to define it before this check. However, if you meant to check the value of `$("#Title").val()` instead, update the condition accordingly.

### 4. **Using `return` Outside the Function:**
   - **Issue:** The last `return` statement is outside the `checkTitle()` function. This will cause an error because `return` can only be used inside a function.
   - **Solution:** Make sure the `return checkTitle();` is inside a function, or place it inside a script where it's correctly scoped.

### **Final Revised Code:**

```javascript
function checkTitle() {
  // Select the Title element and get its value
  var titleValue = $("#Title").val();

  // Check if the title is an empty string
  if (typeof titleValue === "string" && titleValue.length === 0) {
    return $("#Format").val() + " - " + $("#Platform").val() + " - ";
  } else if (titleValue === null) {
    return $("#Format").val() + " - " + $("#Platform").val() + " 2 ";
  } else {
    return $("#Format").val() + " - " + $("#Platform").val() + " 3 ";
  }
}

// Call the function and return its result
return checkTitle();
```

### **Summary:**
- **Use `$("#Title").val()`** instead of `$("Title")` to get the value of the input field.
- **Remove the undefined `str` variable** and replace it with the correct logic.
- Ensure **return statements** are inside the correct function or scope.

This should resolve the errors in your code.

**Reply by ConorWebb96** [reply]
Hey @ashishkumarpalai,

I appreciate the effort in responding, but I think it would be more helpful to address Budibase specific questions with accurate and relevant information. The response you provided is quite generic and seemingly AI generated. This may not directly help the user resolve their issue, which could lead to confusion or wasted time.

It might be worth familiarising yourself with Budibase's functionality before answering these types of questions. For example, the `$("Title")` syntax isn't related to jQuery but is a Budibase specific way to reference column values from data sources.

Again thank you for responding!
